Feature - busola extension of btpoperator (#1316)
kyma-project/btp-manager: busola-tests/README.md

> # BTP Operator Busola Extension E2E Tests

Reviewer: ```suggestion
# SAP BTP Operator Busola Extension E2E Tests
```

> Cypress E2E tests for the BTP Operator Busola extension. Tests run against a local k3d cluster and a locally built Busola instance — no external cluster or credentials required.

Reviewer: ```suggestion
Cypress E2E tests for the SAP BTP Operator Busola extension. Tests run against a local k3d cluster and a locally built Busola instance — no external cluster or credentials required.
```

|--------|-------------|
| `--busola-path PATH` | Path to local Busola repository (default: `../busola`) |
| `--skip-cluster` | Reuse existing k3d cluster `kyma` |
| `--skip-busola` | Skip Busola build/start (Busola must already be running on `:3001`) |
| `--headed` | Run Cypress with a visible browser window |
| `--interactive` | Open Cypress GUI for manual test selection |
| `--cleanup` | Delete k3d cluster after tests complete |

## Develop Tests Iteratively

For iterative development, run a full setup once and reuse the cluster and Busola instance on subsequent runs:

```shell
# First run — sets everything up and opens Cypress GUI
./busola-tests/run-local-test.sh --interactive

# Subsequent runs — skip cluster and Busola setup
./busola-tests/run-local-test.sh --skip-cluster --skip-busola --headed
```

To test extension changes without re-running the full suite, apply the updated ConfigMap to the running cluster and rerun:

```shell
kubectl apply -f config/busola-extension/sap-btp-operator-extension.yaml -n kube-public
./busola-tests/run-local-test.sh --skip-cluster --skip-busola --headed
```

## Test Scenarios

All scenarios are in `ext-test-btp-operator.spec.js`.

> **1. Upload Extension ConfigMap**

Reviewer: ```suggestion
### Upload Extension ConfigMap
```

Uploads the extension ConfigMap, verifies the **BTP Operators** menu appears, opens the `btpoperator` detail view, and validates:

> - The Metadata card shows the Documentation link, Service Instances count, and Service Bindings count

Reviewer: ```suggestion
- The Metadata card shows the **Documentation** link, **Service Instances** count, and **Service Bindings** count
```

Reviewer: If these should be treated as UI elements, they should be boldfaced.
If not, let's use natural language (without capitalization): The Metadata card shows the documentation link, service instances count, and service bindings count

Author: they are referring to UI elements.

> - The **BTP Operator Secrets** panel renders with **BTP Manager Secret** (Managed badge) and **SAP BTP Service Operator Secret** (Inherited badge)

Reviewer: ```suggestion
- The **SAP BTP Operator Secrets** panel renders with **BTP Manager Secret** (Managed badge) and **SAP BTP Service Operator Secret** (Inherited badge)
```

- Credentials Namespace defaults to `kyma-system`

> - The **Edit** ResourceLink navigates to the `sap-btp-manager` secret

Reviewer: ```suggestion
- The **Edit** ResourceLink navigates to the `sap-btp-manager` Secret
```

> - The Service Instances and Service Bindings count links navigate to the respective CRD pages

Reviewer: ```suggestion
- The service instances and service bindings count links navigate to the respective CRD pages
```

> **2. Configure Custom Credentials Namespace**

Reviewer: ```suggestion
###  Configure Custom Credentials Namespace
```

> Creates a `test` namespace with a namespace-based secret, edits `sap-btp-manager` to add the `kyma-project.io/skip-reconciliation` label and set `credentials_namespace: test`, then validates:

Reviewer: ```suggestion
Creates a `test` namespace with a namespace-based secret, edits `sap-btp-manager` to add the `kyma-project.io/skip-reconciliation` label and set `credentials_namespace: test`, then validates the following:
```

> - The **BTP Manager Secret** badge switches to Unmanaged

Reviewer: ```suggestion
- The SAP BTP Manager Secret badge switches to **Unmanaged**
```

> - Credentials Namespace shows `test`

Reviewer: ```suggestion
- **Credentials Namespace** shows `test`
```

> - The **Namespace-Based Secrets** table shows the test secret with In Use status

Reviewer: ```suggestion
- The **Namespace-Based Secrets** table shows the test secret with `In Use` status
```

> - After uploading a ServiceInstance and ServiceBinding, counts update in the header

Reviewer: This line isn't quite clear to me.

Author: after uploading si and sb the count should be increased. Maybe we can write it differently?

> **3. Custom Secrets**

Reviewer: ```suggestion
### Custom Secrets
```

> Creates a ServiceInstance with `spec.btpAccessCredentialsSecret` set, then validates:

Reviewer: ```suggestion
Creates a ServiceInstance with `spec.btpAccessCredentialsSecret` set, and validates the following:
```

> - The **Custom Secrets** panel renders with the referenced secret

Reviewer: ```suggestion
- The **Custom Secrets** panel renders with the referenced Secret
```

> - Status shows Not in Use (the secret namespace differs from the credentials namespace)

Reviewer: ```suggestion
- Status shows `Not in Use` (the Secret namespace differs from the credentials namespace)
```

> - The Service Instances count is correct

Reviewer: ```suggestion
- The service instances count is correct
```

## CI

Tests run automatically on pull requests that modify `config/busola-extension/**`, `busola-tests/**`, `config/crd/**`, `examples/btp-operator.yaml`, or the workflow file itself.

> See [`.github/workflows/btp-operator-e2e.yaml`](../.github/workflows/btp-operator-e2e.yaml) for the full workflow definition. On failure, Cypress videos, screenshots, and Busola logs are uploaded as artifacts with a 7-day retention period.

Reviewer: ```suggestion
For the full workflow definition, see [`.github/workflows/btp-operator-e2e.yaml`](../.github/workflows/btp-operator-e2e.yaml). On failure, Cypress videos, screenshots, and Busola logs are uploaded as artifacts with a 7-day retention period.
```
